includes/pipeline_snapshot.js

```diff
@@ -0,0 +1,60 @@
+module.exports = (version, params) => {
+    if ((!params.enableMonitoring) || dataform.projectConfig.schemaSuffix) {
+        /* Don't send pipeline snapshot if monitoring is disabled (enableMonitoring is true by default)
+        or if we're in a Dataform development workspace (schemaSuffix is null if we're in a development workspace) */
+        return true;
+    }
+    return operate("pipeline_snapshot_" + params.eventSourceName, ctx => [`
+BEGIN
+INSERT \`cross-teacher-services.monitoring.pipeline_snapshots\` (
+    workflow_executed_at,
+    gcp_project_name,
+    event_source_name,
+    output_dataset_name,
+    dfe_analytics_version,
+    dfe_analytics_dataform_version,
+    checksum_enabled,
+    number_of_tables,
+    number_of_tables_with_matching_checksums,
+    number_of_rows,
+    number_of_missing_rows,
+    number_of_extra_rows,
+    dfe_analytics_dataform_parameters
+    )
+WITH
+  entity_table_check_scheduled_metrics AS (
+  SELECT
+    COUNT(DISTINCT entity_table_name) > 0 AS checksum_enabled,
+    COUNT(DISTINCT entity_table_name) AS number_of_tables,
+    COUNT(DISTINCT CASE WHEN database_checksum = bigquery_checksum THEN entity_table_name END) AS number_of_tables_with_matching_checksums,
+    SUM(database_row_count) AS number_of_rows,

```

Comment:
Changes look good. Should this also be changed to bigquery_row_count instead of database_row_count

Reply:
I didn't want to because the database count is correct and the BQ isn't - so missing/extra are opinionated relative to the database count

Reply:
right that makes sense - all look good to me